just-kdl/src/dom.rs

```diff
@@ -174,17 +178,18 @@ impl<'text> Node<'text> {
 		let mut seen = HashSet::new();
 		for entry in self.entries.iter_mut().rev() {
 			if let Some(key) = &mut entry.key {
-				if seen.contains(key) {
+				if seen.contains(&**key) {
 					*key = Cow::Borrowed(marker);
 				} else {
-					seen.insert(&*key);
+					seen.insert(&**key);
 				}
 			}
 		}
+		drop(seen);
```

Comment:
Genuinely, I'm not sure what's different between the API for `hashbrown::HashSet` and `std::collections::HashSet` that requires this, but, it fixed the error, so 🤷🏻

The above extra `&**key` was me trying to make the `HashSet` have `&str` keys which didn't seem to help.